snuba/query/parser/expressions.py

```diff
@@ -62,44 +86,90 @@ def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> st
     def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
         return Column(None, None, node.text)
 
-    def visit_empty(
+    def visit_low_pri_tuple(
+        self, node: Node, visited_children: Tuple[LowPriOperator, Expression]
+    ) -> LowPriTuple:
+        left, right = visited_children
+
+        return LowPriTuple(op=left, arithm=right)
+
+    def visit_high_pri_tuple(
+        self, node: Node, visited_children: Tuple[HighPriOperator, Expression]
+    ) -> HighPriTuple:
+        left, right = visited_children
+
+        return HighPriTuple(op=left, arithm=right)
+
+    def visit_low_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> LowPriOperator:
+
+        return LowPriOperator(node.text)
+
+    def visit_high_pri_op(
         self, node: Node, visited_children: Iterable[Any]
-    ) -> Optional[Expression]:
-        return None
+    ) -> HighPriOperator:
+
+        return HighPriOperator(node.text)
+
+    def visit_arithmetic_term(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any]
+    ) -> Expression:
+        _, term, _ = visited_children
+
+        return term
 
     def visit_low_pri_arithmetic(
         self,
         node: Node,
-        visited_children: Tuple[Any, Expression, Any, Tuple[str, Expression]],
+        visited_children: Tuple[
+            Any, Expression, Any, Union[Node, LowPriTuple, List[LowPriTuple]]
```

Comment:
nit: please replace List with Sequence. It is more abstract.

Reply:
Done.